### PR TITLE
gather_data: fix time_queued calculating

### LINKED ISSUE #137: Bug: filters by runner_labels and job_names works incorrect
When we add possibility to filter data with `runner_labels` and `job_name`, we don't see data about jobs on macOS and jobs with multiply labels. This goes because of parsing error: when we get tag set in Grafana query request, we parse it wrong if the source tag value has a comma in the name.

Data with filter enabled, but `All` set in the filter:
image

Data with filter disabled:
image

### PR DESCRIPTION
**Pull request by lastoCHka42**
Before this patch, there was an error occured on the `gathering_data.py`
script:

```
ValueError: Invalid isoformat string: '\ufeff2024-03-01T06:02:0+00:00'
```

This error occures, if there is a BOM at the file first byte. Before
this patch, we calculated `time_queued` as first 20 symbols of the first
log line. This patch fixes the problem with reqex search the ISO format
date and time.

### CODE REVIEW

File: multivac/gather_data.py [hunk 1]
```diff
@@ -110,6 +111,13 @@ def github_time_to_unix(time: str) -> float:
     time_to_unix = datetime.timestamp(time_to_datetime)
     return time_to_unix
 
+def get_log_datetime(log_line: str) -> str:
+    # Get ISO format date and time from the log line
+    datetime_match = DATETIME_RE.search(log_line)
+    if datetime_match:
+        return f'{datetime_match.group(0)}Z'
+    print(f"NO TIME IN {log_line}")
```
Comment (ochaplashkin) [inline multivac/gather_data.py:119]: Delete the debug output.

File: multivac/gather_data.py [hunk 2]
```diff
@@ -318,7 +326,8 @@ def sorter(filename):
             except FileNotFoundError:
                 print(f'No logs for job {job_id}, {job["html_url"]}')
             else:
-                time_queued = log_file_as_list[0][0:19] + 'Z'
+                time_queued = get_log_datetime(log_file_as_list[0]) \
```
Comment (ochaplashkin) [inline multivac/gather_data.py:329]: Add a comment on why the value is passed at 0, and not just as is (list)

**Review by ochaplashkin — CHANGES_REQUESTED** [review]

**Comment by ochaplashkin** [comment]
Also, check code-style (flake8 has been failed)

File: multivac/gather_data.py (hunk 1 above)
Comment (lastoCHka42) [reply multivac/gather_data.py]: Done

File: multivac/gather_data.py (hunk 2 above)
Comment (lastoCHka42) [reply multivac/gather_data.py:331]: Done

**Review by ochaplashkin — APPROVED** [review]
LGTM :rocket:

**Pull request merged**